src/main/java/panav/task/Task.java

```diff
@@ -0,0 +1,51 @@
+package panav.task;
+
+/**
+ * Abstract class to represent a Task in the TaskList.
+ */
+public abstract class Task {
+
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Method to show if a task is completed or not.
+     * @return "X" is task is done, " " otherwise.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```

Comment:
Great code quality shown! The choice of conditionals over if-else statements make your code looks cleaner!